### PR TITLE
Fix ndg break

### PR DESCRIPTION
**Pull request by iurisilvio**

### CODE REVIEW

**Comment by iurisilvio** [comment]
I wrote tests to #289.

I created another tox env to reproduce the issue. If you just run the tests with `ndg-httpsclient>=0.4` installed, they fail.

```
   AssertionError: Traceback (most recent call last):
     File "/home/iurisilvio/code/oss/HTTPretty/.tox/py27-ndg/local/lib/python2.7/site-packages/sure/__init__.py", line 186, in wrap
       func(start, *args, **kw)
     File "/home/iurisilvio/code/oss/HTTPretty/tests/functional/test_requests.py", line 316, in test_streaming_responses
       auth=('username', 'password'), stream=True)
     File "HTTPretty/.tox/py27-ndg/local/lib/python2.7/site-packages/requests/api.py", line 111, in post
       return request('post', url, data=data, json=json, **kwargs)
     File "HTTPretty/.tox/py27-ndg/local/lib/python2.7/site-packages/requests/api.py", line 57, in request
       return session.request(method=method, url=url, **kwargs)
     File "HTTPretty/.tox/py27-ndg/local/lib/python2.7/site-packages/requests/sessions.py", line 475, in request
       resp = self.send(prep, **send_kwargs)
     File "HTTPretty/.tox/py27-ndg/local/lib/python2.7/site-packages/requests/sessions.py", line 585, in send
       r = adapter.send(request, **kwargs)
     File "HTTPretty/.tox/py27-ndg/local/lib/python2.7/site-packages/requests/adapters.py", line 477, in send
       raise SSLError(e, request=request)
   SSLError: ("bad handshake: SysCallError(32, 'EPIPE')",)
   ```

File: httpretty/core.py [hunk 1]
```diff
@@ -102,6 +102,13 @@
 except ImportError:  # pragma: no cover
     ssl = None
 
+# used to handle error caused by ndg-httpsclient
+try:
+    from requests.packages.urllib3.contrib.pyopenssl import inject_into_urllib3, extract_from_urllib3
+    pyopenssl_override = True
+except:
```
Comment (gabrielfalcao) [inline httpretty/core.py]: what is the exact exception to be captured here?

Comment (iurisilvio) [reply httpretty/core.py]: Based on this PR, looks like an `SSLError`, but I did it long time ago. Don't have time to investigate it again.

**Comment by gabrielfalcao** [comment]
thanks @iurisilvio 
closed through c116d7d843a620aff26e4b63f0b8f8f728486800

**Pull request closed without merging**